### PR TITLE
Feature notification service 

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: registration-processor/registration-processor-notification-service-impl/src/main/java/io/mosip/registration/processor/notification/service/impl/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+	private void init() {
+		SubscriptionChangeRequest subscriptionRequest = new SubscriptionChangeRequest();
+		subscriptionRequest.setCallbackURL(callbackURL);
+		subscriptionRequest.setHubURL(hubURL);
+		subscriptionRequest.setSecret(secret);
+		subscriptionRequest.setTopic(topic);
+		sb.subscribe(subscriptionRequest);
+	}
+
+	@PostMapping(value = "/callback", consumes = "application/json")
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-notification-service-impl/src/main/java/io/mosip/registration/processor/notification/service/impl/NotificationService.java:151]: Move the request mappings to a controller layer and have only the business logic in this service

File: registration-processor/registration-processor-notification-service-impl/src/main/resources/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+registration.processor.demographic.identity=identity
+packet.info.storage.service=registration-processor-packet-info-storage-service
+config.server.file.storage.uri=${spring.cloud.config.uri}/${packet.info.storage.service}/${spring.profiles.active}/${spring.cloud.config.label}/
+
+server.port=8088
+health.config.enabled=false
+management.endpoints.web.exposure.include=refresh
+eventbus.port=5721
+registration.processor.zone=secure
+vertx.cluster.configuration=${spring.cloud.config.uri}/*/${spring.profiles.active}/${spring.cloud.config.label}/hazelcast_${registration.processor.zone}.xml
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-notification-service-impl/src/main/resources/bootstrap.properties:16]: Why do we need vertx and kafka configuration here? Remove all unwanted configs

File: registration-processor/registration-processor-notification-service-impl/src/test/java/io/mosip/registration/processor/notification/service/test/NotificationServiceTestApplication.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Audit manager application
+ * 
+ * @author Dharmesh Khandelwal
+ * @since 1.0.0
+ *
+ */
+@SpringBootApplication(scanBasePackages =  { "io.mosip.registration.processor.notification.service.*"},exclude = {HibernateDaoConfig.class} )
+public class NotificationServiceTestApplication {
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-notification-service-impl/src/test/java/io/mosip/registration/processor/notification/service/test/NotificationServiceTestApplication.java:18]: What does this class do ?

File: registration-processor/registration-processor-notification-service-impl/src/test/resources/bootstrap.properties [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+registration.processor.demographic.identity=identity
+packet.info.storage.service=registration-processor-packet-info-storage-service
+config.server.file.storage.uri=${spring.cloud.config.uri}/${packet.info.storage.service}/${spring.profiles.active}/${spring.cloud.config.label}/
+
+server.port=8088
+health.config.enabled=false
+management.endpoints.web.exposure.include=refresh
+eventbus.port=5721
+registration.processor.zone=secure
+vertx.cluster.configuration=${spring.cloud.config.uri}/*/${spring.profiles.active}/${spring.cloud.config.label}/hazelcast_${registration.processor.zone}.xml
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-notification-service-impl/src/test/resources/bootstrap.properties:16]: Remove all unwanted properties

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-notification-service-impl/src/test/java/io/mosip/registration/processor/notification/service/test/NotificationServiceTestApplication.java (hunk 3 above)
Comment (urviljoshi) [reply registration-processor/registration-processor-notification-service-impl/src/test/java/io/mosip/registration/processor/notification/service/test/NotificationServiceTestApplication.java:18]: This is a test app class this class just create context when running test junit

**Pull request closed without merging**